Framework/PythonInterface/plugins/algorithms/HB3AAdjustSampleNorm.py

```diff
@@ -0,0 +1,272 @@
+        # Make sure files and workspaces aren't both set
+        if len(filelist) >= 1:
+            if not input_ws.isDefault:
+                issues['InputWorkspaces'] = "Cannot specify both a filename and input workspace"
+        else:
+            if input_ws.isDefault:
+                issues['Filename'] = "Either a file or input workspace must be specified"
+
+        if len(vanfile) > 0 and not van_ws.isDefault:
+            issues['VanadiumWorkspace'] = "Cannot specify both a vanadium file and workspace"
+
+        # Verify given workspaces exist
+        if not input_ws.isDefault:
+            input_ws_list = list(map(str.strip, input_ws.value.split(",")))
+            for ws in input_ws_list:
+                if not mtd.doesExist(ws):
+                    issues['InputWorkspaces'] = "Could not find input workspace '{}'".format(ws)
+                else:
+                    # If it does exist, make sure the workspace is an MDHisto with 3 dimensions
+                    if not isinstance(mtd[ws], IMDHistoWorkspace):
+                        issues['InputWorkspaces'] = "Workspace '{}' must be a MDHistoWorkspace".format(ws)
+                    elif mtd[ws].getNumDims() != 3:
+                        issues['InputWorkspaces'] = "Workspace '{}' expected to have 3 dimensions".format(ws)
+
+        return issues
+
+    def PyExec(self):
+        load_van = not self.getProperty("VanadiumFile").isDefault
+        load_files = not self.getProperty("Filename").isDefault
+
+        has_van = not self.getProperty("VanadiumWorkspace").isDefault and not load_van
+
+        if load_files:
+            datafiles = self.getProperty("Filename").value
+        else:
+            datafiles = list(map(str.strip, self.getProperty("InputWorkspaces").value.split(",")))
+
+        vanadiumfile = self.getProperty("VanadiumFile").value
+        vanws = self.getProperty("VanadiumWorkspace").value
+        height = self.getProperty("DetectorHeightOffset").value
+        distance = self.getProperty("DetectorDistanceOffset").value
+        method = self.getProperty("OutputAsMDEventWorkspace").value
+
+        wavelength = -1
+        # Set the back-up wavelength value
+        wl_prop = self.getProperty("Wavelength")
+        if not wl_prop.isDefault:
+            wavelength = wl_prop.value
+
+        if method:
+            minvals = self.getProperty("MinValues").value
+            maxvals = self.getProperty("MaxValues").value
+            merge = self.getProperty("MergeInputs").value
+            wslist = []
+        else:
+            bin0 = self.getProperty("BinningDim0").value
+            bin1 = self.getProperty("BinningDim1").value
+            bin2 = self.getProperty("BinningDim2").value
+
+        out_ws = self.getPropertyValue("OutputWorkspace")
+        out_ws_name = out_ws
+
+        if load_van:
+            vanws = LoadMD(vanadiumfile, StoreInADS=False)
+
+        has_multiple = True if len(datafiles) > 1 else False
+
+        for file in datafiles:
```

Comment:
`file` is a build-in type, you should use a different name so you aren't overriding it.